core/trino-parser/src/main/java/io/trino/sql/tree/NodeRef.java

```diff
@@ -58,9 +57,6 @@ public int hashCode()
     @Override
     public String toString()
     {
-        return format(
-                "@%s: %s",
-                Integer.toHexString(identityHashCode(node)),
-                node);
+        return "@" + Integer.toHexString(identityHashCode(node)) + ": " + node;
```

Comment:
Similar comment here. This shouldn't matte from a performance perspective.

Reply:
Since this `toString()` implementation seems to only be present for more helpful debugging, I'm fine with changing it back here and will rebase without this change.

However, at a high level- I think the general philosophy should be to avoid using `String.format` in normal (ie: non-exceptional, not just for debugging) code paths and to prefer string concatenation instead- especially for simple things like `format("%s=%s", ...)`. JDK 9+ `javac` implementations generate more efficient, specialized to the call-site concatenations even compared to `StringBuilder`, and even small inefficiencies can add up to a non-trivial overall performance overhead.